sql/catalyst/src/main/java/org/apache/spark/sql/catalyst/expressions/UnsafeRow.java

```diff
@@ -236,6 +236,41 @@ public Object get(int ordinal) {
   }
 
   @Override
+  public Object get(int ordinal, DataType dataType) {
+    if (dataType instanceof NullType) {
```

Comment:
Actually, it looks like our existing row behavior is to just return the zero-value of the given type for null inputs (e.g. getFloat on a null column returns 0.0f whereas the generic getter returns null).  For some reason, it looks like UnsafeRow was returning NaN instead of 0 in those cases, leading to a confusing bug.  I'm going to fix this inconsistency in a separate patch.